File: discover_javascript/write_them_to_disc.js
var https = require('https');
var fs = require('fs')

var options = {
    hostname: 'api.github.com',
    path: '/search/repositories?q=language:javascript&sort=stars\
&order=desc',
    headers: {
	'User-Agent': 'Holberton_School',
	'Authorization': 'token 58d7730b72d5e9ee5abe4484958359cc\
b67aa53a'
    }
}

var request = https.request(options, function(res) {
    streamToString (res, (jsonString) => {
	console.log("The file was saved!");
	fs.writeFile('/tmp/21', jsonString, function (err) {
	    if (err) return console.log(err);
	});
    });
});
request.end();

request.on('error', function(e) {
    console.error(e);
});

function streamToString(stream, cb) {
    const chunks = [];
    stream.on('data', (chunk) => {
	chunks.push(chunk);
    });
    stream.on('end', () => {
	cb(chunks.join(''));
    });
}

File: discover_more_python/write_them_to_disc.py
import requests
# Function that gets githubs popular python repos and writes json to file.
request_headers = {
  'User-Agent': 'Holberton_School',
  'Authorization': 'token b28c7b657c6fd8ba06b3d8bb7bf1835d1f50b40f'
}

r = requests.get('https://api.github.com/search/repositories?q=language:python&sort=stars&order=desc', request_headers)

with open('/tmp/21', 'w') as f:
	f.write(r.content)

print "The file was saved!"
